Source/Tools/LinuxEmulation/LinuxSyscalls/SyscallsSMCTracking.cpp

```diff
@@ -153,6 +153,177 @@ FEXCore::HLE::AOTIRCacheEntryLookupResult SyscallHandler::LookupAOTIRCacheEntry(
+      if (Result == ~0ULL) {
+        return reinterpret_cast<void*>(-errno);
+      }
+    }
+
+    FEX::HLE::_SyscallHandler->TrackMmap(Thread, Result, length, prot, flags, fd, offset);
+  }
+
+  FEX::HLE::_SyscallHandler->InvalidateCodeRangeIfNecessary(Thread, Result, Size);
+  return reinterpret_cast<void*>(Result);
+}
+
+uint64_t SyscallHandler::GuestMunmap(bool Is64Bit, FEXCore::Core::InternalThreadState* Thread, void* addr, uint64_t length) {
+  LOGMAN_THROW_A_FMT(Is64Bit || (reinterpret_cast<uintptr_t>(addr) >> 32) == 0, "values must fit to 32 bits: {}", fmt::ptr(addr));
+  LOGMAN_THROW_A_FMT(Is64Bit || (length >> 32) == 0, "values must fit to 32 bits");
+
+  uint64_t Result {};
+  uint64_t Size = FEXCore::AlignUp(length, FEXCore::Utils::FEX_PAGE_SIZE);
+
+  {
+    // Frontend calls this with nullptr Thread during initialization.
+    // This is why `GuardSignalDeferringSectionWithFallback` is used here.
+    // To be more optimal the frontend should provide this code with a valid Thread object earlier.
+    auto lk = FEXCore::GuardSignalDeferringSectionWithFallback(VMATracking.Mutex, Thread);
+
+    if (reinterpret_cast<uintptr_t>(addr) < 0x1'0000'0000ULL) {
+      Result = FEX::HLE::_SyscallHandler->Get32BitAllocator()->Munmap(addr, length);
+      if (FEX::HLE::HasSyscallError(Result)) {
+        return Result;
+      }
+    } else {
+      Result = ::munmap(addr, length);
+      if (Result == -1) {
+        return -errno;
+      }
+    }
+    FEX::HLE::_SyscallHandler->TrackMunmap(Thread, addr, length);
+  }
+  FEX::HLE::_SyscallHandler->InvalidateCodeRangeIfNecessary(Thread, reinterpret_cast<uint64_t>(addr), Size);
+
+  return Result;
+}
+
+uint64_t SyscallHandler::GuestMremap(bool Is64Bit, FEXCore::Core::InternalThreadState* Thread, void* old_address, size_t old_size,
+                                     size_t new_size, int flags, void* new_address) {
+  uint64_t Result {};
+
+  {
+    auto lk = FEXCore::GuardSignalDeferringSection(FEX::HLE::_SyscallHandler->VMATracking.Mutex, Thread);
+    if (Is64Bit) {
+      Result = reinterpret_cast<uint64_t>(::mremap(old_address, old_size, new_size, flags, new_address));
+      if (Result == -1) {
+        return Result;
+      }
+    } else {
+      Result =
+        reinterpret_cast<uint64_t>(FEX::HLE::_SyscallHandler->Get32BitAllocator()->Mremap(old_address, old_size, new_size, flags, new_address));
+      if (FEX::HLE::HasSyscallError(Result)) {
+        return Result;
+      }
+    }
+    FEX::HLE::_SyscallHandler->TrackMremap(Thread, reinterpret_cast<uint64_t>(old_address), old_size, new_size, flags, Result);
+  }
+
+  FEX::HLE::_SyscallHandler->InvalidateCodeRangeIfNecessaryOnRemap(Thread, reinterpret_cast<uint64_t>(old_address), Result, old_size, new_size);
+  return Result;
+}
+
+uint64_t SyscallHandler::GuestMprotect(FEXCore::Core::InternalThreadState* Thread, void* addr, size_t len, int prot) {
```

Comment:
`GuestMprotect` is the only function out of these seven `Guest*` that don't return a combined result+errno value instead keeping it separate. This is a weird contrast and instead it should just return -errno on error and remove the `SYSCALL_RET()` in the handlers.

This makes all the "Guest" handlers return a combined result for consistency.